pkg/apis/apps/v1alpha1/apimanager_types.go

```diff
@@ -246,6 +317,76 @@ func (apimanager *APIManager) setApicastSpecDefaults() bool {
 		changed = true
 	}
 
+	if spec.Apicast.ApicastStagingSpec != nil {
+		spec.Apicast.ApicastStagingSpec = &ApicastStagingSpec{}
+		changed = true
+	}
+
+	if spec.Apicast.ApicastProductionSpec != nil {
+		spec.Apicast.ApicastProductionSpec = &ApicastProductionSpec{}
+		changed = true
+	}
+
+	if spec.Apicast.ApicastStagingSpec.Replicas == nil {
+		spec.Apicast.ApicastStagingSpec.Replicas = apimanager.defaultReplicas()
+		changed = true
+	}
+
+	if spec.Apicast.ApicastProductionSpec.Replicas == nil {
+		spec.Apicast.ApicastProductionSpec.Replicas = apimanager.defaultReplicas()
+		changed = true
+	}
+
+	return changed
+}
+
+func (apimanager *APIManager) defaultReplicas() *int64 {
+	var defaultReplicas int64 = 1
+	if apimanager.Spec.HighAvailability != nil && apimanager.Spec.HighAvailability.Enabled {
```

Comment:
I would totally remove replica dependency of `HighAvailability`.

Besides, but independently of removing replica dependency of `HighAvailability`, IMO, we should deprecate `HighAvailability` option. At least call it `DatabaseLess`. Or even better: let `Image` CR fields rule. For instance, if `APIManagerSpec.Backend.RedisImage` is empty or nil, then try to read external redis connection strings from the associated secret.